extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js

```diff
@@ -134,25 +161,55 @@ class AmpNexxtvPlayer extends AMP.BaseElement {
       src += `&streamingFilter=${encodeURIComponent(streamingFilter)}`;
     }
 
-    return assertAbsoluteHttpOrHttpsUrl(src);
-  }
+    if (exitMode !== null && exitMode.length > 0) {
+      src += `&exitMode=${encodeURIComponent(exitMode)}`;
+    }
 
-  /** @override */
-  layoutCallback() {
-    const iframe = createFrameFor(this, this.getVideoIframeSrc_());
+    if (this.consentString_ !== null){
+      src += `&consentString=${encodeURIComponent(this.consentString_)}`;
+    }
 
-    this.iframe_ = iframe;
+    return assertAbsoluteHttpOrHttpsUrl(src);
+  }
 
-    this.unlistenMessage_ = listen(this.win, 'message', (event) => {
-      this.handleNexxMessage_(event);
+  /**
+   * Get consent data from consent module
+   * @return {Promise[]}
+   */
+  getConsents_() {
+    const consentPolicy = super.getConsentPolicy() || 'default';
+    const consentPromise = consentPolicy
+      ? getConsentPolicyState(this.element, consentPolicy)
+      : Promise.resolve(null);
+    const sharedDataPromise = consentPolicy
+      ? getConsentPolicySharedData(this.element, consentPolicy)
+      : Promise.resolve(null);
+    const consentStringPromise = consentPolicy
+      ? getConsentPolicyInfo(this.element, consentPolicy)
+      : Promise.resolve(null);
+
+    return Promise.all([
+      consentPromise,
+      sharedDataPromise,
+      consentStringPromise,
+    ]).then((consentData) => {
```

Comment:
It seems like you're only using `consentString_` above. Do you need get the two other consent values?

Reply:
yea you're right, I initially wanted to use, but forgot to remove.